OfflineUX.md

```diff
@@ -0,0 +1,129 @@
+#### Offline UX Patterns
+
+### Contents
+* [Motivation & Philosophy](#motivation-&-philosophy)
+* [UX Pattern Flowchart](#ux-pattern-flowchart)
+* [How to Answer Questions on the Flow Chart](#how-to-answer-questions-on-the-flowchart)
+* [Description of the Patterns](#description-of-the-patterns)
+    - [None - No Offline Behavior](#none---no-offline-behavior)
+    - [A - Optimistic Without Feedback](#a---optimistic-without-feedback)
+    - [B - Optimistic With Feedback](#b---optimistic-with-feedback)
+    - [C - Blocking Form](#c---blocking-form)
+    - [D - Full Page Blocking](#d---full-page-blocking)
+
+### Motivation & Philosophy
+
+Understanding the offline behavior of our app is vital to becoming a productive contributor to the Expensify codebase. Our mission is to support our users in every possible environment, and often our app is used in places where a stable internet connection is not guaranteed. 
+
+The most important concept to keep in mind while reading this document is that we want to allow users to do as much as possible when offline. At first, this might seem impossible because almost everything the user can touch in our app is related to an API request. However, in many cases, we can save that API request and assume it will succeed when the user is back online. We then allow the user to proceed as if their request already succeeded. We call this an optimistic response. Here, we use the word **optimistic** to indicate that we’re confident the request will succeed when the user is online. 
+
+<div style="margin: 15px; padding: 15px; border: 3px solid grey">
+Example: Pinning a chat
+
+When a user clicks the pin button <img style="height: 10px; width: 10px; filter: invert(.5) saturate(5);" src="./assets/images/pin.svg"/> on a chat, two things should happen. 
+
+We send a request to the API to keep track of that. After all, if the user signed in on a different device, we would want all of their pinned chats to show up (another way of saying this is we want their pinned chats to ‘persist’ across devices). 
+The chat should go to the top of the list with the other pinned chats, and the pin button should look darker than it did before. This is the user’s visual feedback that clicking the pin button worked.
+
+If the user is offline, do we need to wait for the API request to finish before doing all that visual stuff? No! We are optimistic that the API request will succeed once it’s sent. In the meantime, we let the user continue using the app. 
+</div>
+
+The example we just looked at is nice and simple, but what about something more complicated, like requesting money from another user? What about paying another user? Or even worse, removing a user from a workspace? For these types of actions, we can’t simply proceed as if the request already finished. That would have serious consequences for the user. This is why we have developed different ways of handling these cases. These are called the UX Patterns, which are explained below.
```

Comment:
> For these types of actions, we can’t simply proceed as if the request already finished. That would have serious consequences for the user.

This part is leaving a lot of assumptions up to the reader. I think it would be better to be very clear and explicit and explain all assumptions clearly.